plugins/autopeering/local/local.go

```diff
@@ -26,41 +26,55 @@ var (
 func configureLocal() *peer.Local {
 	log := logger.NewLogger("Local")
 
-	var externalIP net.IP
+	var peeringIP net.IP
 	if str := config.NodeConfig.GetString(config.CfgNetAutopeeringExternalAddr); strings.ToLower(str) == "auto" {
-		log.Info("Querying external IP ...")
-		ip, err := netutil.GetPublicIP(config.NodeConfig.GetBool(config.CfgNetPreferIPv6))
-		if err != nil {
-			log.Fatalf("Error querying external IP: %s", err)
-		}
-		log.Infof("External IP queried: address=%s", ip.String())
-		externalIP = ip
+		// let the autopeering discover the IP
+		peeringIP = net.IPv4zero
 	} else {
-		externalIP = net.ParseIP(str)
-		if externalIP == nil {
+		peeringIP = net.ParseIP(str)
+		if peeringIP == nil {
 			log.Fatalf("Invalid IP address (%s): %s", config.CfgNetAutopeeringExternalAddr, str)
 		}
-	}
 
-	if !externalIP.IsGlobalUnicast() {
-		log.Fatalf("IP is not a global unicast address: %s", externalIP.String())
+		if !peeringIP.IsGlobalUnicast() {
+			log.Warnf("IP is not a global unicast address: %s", peeringIP.String())
+		}
 	}
 
-	_, peeringPort, err := net.SplitHostPort(config.NodeConfig.GetString(config.CfgNetAutopeeringBindAddr))
+	_, peeringPortStr, err := net.SplitHostPort(config.NodeConfig.GetString(config.CfgNetAutopeeringBindAddr))
 	if err != nil {
 		log.Fatalf("autopeering bind address is invalid: %s", err)
 	}
 
+	peeringPort, err := strconv.Atoi(peeringPortStr)
+	if err != nil {
+		log.Fatalf("Invalid autopeering port number: %s, Error: %s", peeringPortStr, err)
+	}
+
+	if 0 > peeringPort || peeringPort > 65535 {
+		log.Fatalf("Invalid autopeering port number (%s): %d", config.CfgNetAutopeeringBindAddr, peeringPort)
+	}
+
 	// announce the peering service
 	ownServices := service.New()
-	ownServices.Update(service.PeeringKey, "udp", net.JoinHostPort(externalIP.String(), peeringPort))
+	ownServices.Update(service.PeeringKey, "udp", peeringPort)
+
 	if !config.NodeConfig.GetBool(config.CfgNetAutopeeringRunAsEntryNode) {
-		_, gossipBindAddrPort, err := net.SplitHostPort(config.NodeConfig.GetString(config.CfgNetGossipBindAddress))
+		_, gossipBindAddrPortStr, err := net.SplitHostPort(config.NodeConfig.GetString(config.CfgNetGossipBindAddress))
```

Comment:
What happens when the IP part of `config.CfgNetGossipBindAddress` is different from  the IP part of `config.CfgNetAutopeeringBindAddr`? I guess, currently nothing not even a warning. Is this expected? Maybe make only the gossip port configurable?